napari/_qt/containers/_layer_delegate.py

```diff
@@ -112,6 +123,26 @@ def get_layer_icon(
         option.decorationPosition = option.Right  # put icon on the right
         option.features |= option.HasDecoration
 
+    def _paint_loading(
+        self,
+        painter: QPainter,
+        option: QStyleOptionViewItem,
+        index: QtCore.QModelIndex,
+    ):
+        loaded = index.data(LoadedRole)
+        if not loaded:
+            load_rect = option.rect.translated(55, 8)
+            h = index.data(Qt.ItemDataRole.SizeHintRole).height() - 16
+            load_rect.setWidth(h)
+            load_rect.setHeight(h)
+
+            if self.load_movie.state() == QMovie.NotRunning:
+                self.load_movie.start()
```

Comment:
Based on the videos in the PR description, constantly restarting the loading animation is visually jarring. I'd expect there to be some delay before the animation starts (e.g 1 second). Could we achieve that with a single shot timer? Do we need something more complex to handle multiple calls to start? Or does `QMovie` have some built-in functionality for a delayed start?

Reply:
I'm not sure if `QMovie` has something built-in that could help with this but I will check. Thank you for the feedback!

Reply:
That was my first reaction as well. I don't knwo what a "good" delay is, but let's give it a go with 1 second and see how it feels when actually interacting with the viewer.

Reply:
Checking, instead of stopping the animation we can paused it, which makes things look something like this:

![loading_paused](https://user-images.githubusercontent.com/16781833/202525810-5244b9db-c2d8-460f-bccc-6afc13048ed1.gif)

Anyhow, I will continue checking for a way to start the animation with some sort of delay

Reply:
Any idea why is the loading dots animation smooth only when you move around the canvas, else blinky?

Reply:
Thanks for the clear explanation. What got me I guess is that the animation seemed to proceed, rather than restarting per slice.

Reply:
Thanks for the question (actually it helped me realize the effect of the delay :+1:)! And so you think it would be better to just restart the animation per slice (as in the initial GIF at https://github.com/napari/napari/pull/5342#issue-1451813387), right?

I'm checking locally things to make the animation less visually jarring and since the animation can be paused I was testing that too, but originally I was thinking in the animation being fully restarted per loaded even. Maybe just adding the delay could be enough?

Reply:
I think with the delay it's more clear what is happening? 🤷‍♂️

Reply:
What if there was an always-visible indicator that just proceeds along the animation when necessary? I still don't love the jumpy result even after 1.5 delay :/

Reply:
looks 🔥 to me

Reply:
Yeah, way better, nice!